flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -199,7 +207,7 @@ private static HiveConf createHiveConf(@Nullable String hiveConfDir) {
 		Configuration hadoopConf = HadoopUtils.getHadoopConfiguration(new org.apache.flink.configuration.Configuration());
 
 		// Add mapred-site.xml. We need to read configurations like compression codec.
-		for (String possibleHadoopConfPath : HadoopUtils.possibleHadoopConfPaths(new org.apache.flink.configuration.Configuration())) {
+		for (String possibleHadoopConfPath : HadoopUtils.possibleHadoopConfPaths(new org.apache.flink.configuration.Configuration(), hadoopConfDir)) {
```

Comment:
Let's not modify `HadoopUtils`. Instead, if `hadoopConfDir` is not null, set `ConfigConstants.PATH_HADOOP_CONFIG` in the `Configuration` instance.